src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -166,4 +168,192 @@ public static String getCollection(Firestore database, String collectionToGetFro
        throw new FirebaseException(ServletConstantValues.UNABLE_TO_READ_FROM_FIRESTORE, e);
     }
   }
-} 
\ No newline at end of file
+
+  /**
+   * Adds support for infinite scroll by batching queries into pages of a limited size.
+   * Either we send in the first requested batch, or we send the next batch after one has 
+   * already been sent.
+   * @param database - The database to retrieve from.
+   * @param collectionToGetFrom - The name of the collection to retrieve the batch from.
+   * @param batchSizeLimit - The maximum size of the batch, which is the size of a new page of 
+   *   data in infinite scroll.
+   * @param lastDocID - The ID of the last object I sent from the most recent batch,
+   *   if null, then execute getFirstBatch().
+   * @param parameterToSortBy - The parameter I want to presort my collection by, 
+   *   if null then don't sort.
+   * @return - The json string representing the new batch of documents to be sent to the frontend.
+   * @throws FirebaseException
+   */
+  public static String getCollectionBatch(Firestore database, String collectionToGetFrom, 
+      int batchSizeLimit,  String lastDocID, String parameterToSortBy) 
+      throws FirebaseException {
+    // A batch size limit needs to be specified in order to make any query.
+    if (batchSizeLimit == 0) {
+      throw new FirebaseException(BATCH_SIZE_NOT_SPECIFIED);
+    }
+
+    CollectionReference collectionReference = database.collection(collectionToGetFrom);
+
+    // If the lastDocID is not present, then we know this is the first batch to send.
+    // Else simply get the next batch in the collection.
+    Optional<String> optionalID = Optional.ofNullable(lastDocID);
+    if (!optionalID.isPresent()) {
+      return getFirstBatch(collectionReference, batchSizeLimit, parameterToSortBy);
+    } else {
+      return getNextBatch(database, collectionReference, batchSizeLimit, optionalID.get(), parameterToSortBy);
+    }
+  }
+
+
+  /**
+   * Helper method for getCollectionBatch().
+   * Retrieve the first batch in the collection collectionReference of size batchSizeLimit.
+   * @param collectionReference - A reference to the collection I want to retrieve from.
+   * @param batchSizeLimit - The maximum size of the batch, which is the size of a new page of data in infinite scroll.
+   * @param parameterToSortBy - The parameter to sort the list of documents by.
+   * @return - The json string representing the first batch in a request for all of the documents in a collection.
+   * @throws FirebaseException
+   */
+  private static String getFirstBatch(CollectionReference collectionReference, int batchSizeLimit, String parameterToSortBy) 
+      throws FirebaseException {
```

Comment:
So should I directly pass in an optional object for id, batchSize and parameterToSortBy? We need to talk about this later today.